controllers/scopetemplate_controller_test.go

```diff
@@ -0,0 +1,362 @@
+							GenerateName: "test-",
+						},
+					}
+					Expect(k8sClient.Create(ctx, namespace2)).NotTo(HaveOccurred())
+				})
+
+				AfterEach(func() {
+					Expect(k8sClient.Delete(ctx, namespace2)).NotTo(HaveOccurred())
+				})
+
+				It("Should create new roleBindings in the expected namespaces", func() {
+					Eventually(func() error {
+						scopeInstance := &operatorsv1.ScopeInstance{}
+						Expect(k8sClient.Get(ctx, types.NamespacedName{Name: scopeInstanceName}, scopeInstance)).NotTo(HaveOccurred())
+						scopeInstance.Spec.Namespaces = []string{namespace.GetName(), namespace2.GetName()}
+						Expect(k8sClient.Update(ctx, scopeInstance)).NotTo(HaveOccurred())
+						return nil
+					}, timeout, interval).Should(BeNil())
+
+					labels := map[string]string{scopeInstanceUIDKey: string(scopeInstance.GetUID()),
+						scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
+						clusterRoleBindingGenerateKey: "test"}
+
+					roleBindingList := listRoleBinding(namespace2.GetName(), 1, labels)
+
+					existingRB := &roleBindingList.Items[0]
+
+					verifyRoleBindings(existingRB, scopeInstance, scopeTemplate)
+
+					roleBindingList = listRoleBinding(namespace.GetName(), 1, labels)
+
+					existingRB = &roleBindingList.Items[0]
+					verifyRoleBindings(existingRB, scopeInstance, scopeTemplate)
+				})
+
+				When("a scopeInstance is updated to remove one of the namespace", func() {
+					It("Should remove respective roleBindings in the expected namespaces", func() {
+						Eventually(func() error {
+							scopeInstance := &operatorsv1.ScopeInstance{}
+							Expect(k8sClient.Get(ctx, types.NamespacedName{Name: scopeInstanceName}, scopeInstance)).NotTo(HaveOccurred())
+							scopeInstance.Spec.Namespaces = []string{namespace2.GetName()}
+							Expect(k8sClient.Update(ctx, scopeInstance)).NotTo(HaveOccurred())
+							return nil
+						}, timeout, interval).Should(BeNil())
+
+						labels := map[string]string{scopeInstanceUIDKey: string(scopeInstance.GetUID()),
+							scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
+							clusterRoleBindingGenerateKey: "test"}
+
+						roleBindingList := listRoleBinding(namespace2.GetName(), 1, labels)
+
+						existingRB := &roleBindingList.Items[0]
+
+						verifyRoleBindings(existingRB, scopeInstance, scopeTemplate)
+
+						roleBindingList = listRoleBinding(namespace.GetName(), 0, labels)
+
+						Expect(len(roleBindingList.Items)).To(Equal(0))
+					})
+
+					When("a scopeInstance is updated to remove all namespaces", func() {
+						It("Should create new clusterRoleBindings in the expected namespaces", func() {
+							Eventually(func() error {
+								scopeInstance := &operatorsv1.ScopeInstance{}
+								Expect(k8sClient.Get(ctx, types.NamespacedName{Name: scopeInstanceName}, scopeInstance)).NotTo(HaveOccurred())
+								scopeInstance.Spec.Namespaces = []string{}
+								Expect(k8sClient.Update(ctx, scopeInstance)).NotTo(HaveOccurred())
```

Comment:
No `Expect` funcs in `Eventually` funcs.

Reply:
updated